__tests__/endpoints.test.js

```diff
@@ -262,4 +262,81 @@ describe("POST /api/articles/:article_id/comments", () => {
             expect(body.msg).toBe("Bad Request");
         });
     });
-});
\ No newline at end of file
+});
+
+describe("PATCH /api/articles/:article_id", () => {
+    test("takes key of inc_votes and increments votes on article by that number, returning the updated article", () => {
+        const update = {inc_votes: 1}
+        
+        return request(app)
+        .patch("/api/articles/1")
+        .send(update)
+        .expect(201)
+        .then(({body}) => {
+            expect(body.article).toEqual({
+                title: "Living in the shadow of a great man",
+                topic: "mitch",
+                author: "butter_bridge",
+                body: "I find this existence challenging",
+                created_at: "2020-07-09T20:11:00.000Z",
+                votes: 101,
+                article_id: 1,
+                article_img_url:
+                "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+              })
+        })
+        
+    })
+    test("takes key of inc_votes and correctly updates vote key when inc_votes is a negative number", () => {
+        const update = {inc_votes: -99}
+        
+        return request(app)
+        .patch("/api/articles/1")
+        .send(update)
+        .expect(201)
+        .then(({body}) => {
+            expect(body.article).toEqual({
+                title: "Living in the shadow of a great man",
+                topic: "mitch",
+                author: "butter_bridge",
+                body: "I find this existence challenging",
+                created_at: "2020-07-09T20:11:00.000Z",
+                votes: 1,
+                article_id: 1,
+                article_img_url:
+                "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+              })
+        })
+    })
+    test("404: when trying to update votes on an article that doesnt exist", () => {
+        const update = {inc_votes: 3}
+        return request(app)
+        .patch("/api/articles/34")
+        .send(update)
+        .expect(404)
+        .then(({body}) => {
+            expect(body.msg).toBe("Not Found");
+        });
+    });
+       test("400: Bad request when key isn't inc_votes", () => {
+        const update = {notAkey: 3}
+        return request(app)
+        .patch("/api/articles/2")
+        .send(update)
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe("Bad Request");
+        });
+    });
+        test("400: Bad request when too many keys in request", () => {
+        const update = {inc_votes: 3,
+                        notAkey: 5}
+        return request(app)
+        .patch("/api/articles/2")
+        .send(update)
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe("Bad Request");
+        });
+    });
```

Comment:
This is an interesting test, but it shouldnt give back a 400 error. In this case we should still expect a 200 but the other properties should just be ignored.